scripts/speaker_tasks/create_librispeech_alignment_manifest.py

```diff
@@ -0,0 +1,121 @@
+
+
+def main():
+    input_manifest_filepath = args.input_manifest_filepath
+    base_alignment_path = args.base_alignment_path
+    output_path = args.output_path
+    dataset = args.dataset
+
+    manifest = read_manifest(input_manifest_filepath)
+    target_manifest = []
+    unaligned_path = os.path.join(base_alignment_path, "unaligned.txt")
+    unaligned = get_unaligned_examples(unaligned_path, dataset)
+
+    # separate indices to manage source/destination manifest to handle missing alignments
+    src_i = 0
+    target_i = 0
+    while src_i < len(manifest):
+        file = manifest[src_i]
+        fn = file['audio_filepath'].split('/')[-1]
+        speaker_id = fn.split('-')[0]
+        book_id = fn.split('-')[1]
+
+        book_dir = os.path.join(base_alignment_path, "LibriSpeech", dataset, speaker_id, book_id)
+        alignment_fpath = os.path.join(book_dir, f"{speaker_id}-{book_id}.alignment.txt")
+
+        # Parse each utterance present in the file
+        alignment_file = open(alignment_fpath, "r")
+        for line in alignment_file:
+
+            # filter out unaligned examples
+            file = manifest[src_i]
+            fn = file['audio_filepath'].split('/')[-1]
+            line_id = fn.split('.')[0]
+            while line_id in unaligned:
+                src_i += 1
+                file = manifest[src_i]
+                fn = file['audio_filepath'].split('/')[-1]
+                line_id = fn.split('.')[0]
+
+            # from https://github.com/CorentinJ/librispeech-alignments/blob/master/parser_example.py
+            # Retrieve the utterance id, the words as a list and the end_times as a list
+            utterance_id, words, end_times = line.strip().split(' ')
+            if utterance_id != line_id:
+                raise Exception("Mismatch between source and target utterance id")
+            words = words.replace('\"', '').lower().split(',')
+            end_times = [float(e) for e in end_times.replace('\"', '').split(',')]
+
+            # get speaker ID
+            fn = file['audio_filepath'].split('/')[-1]
+            speaker_id = fn.split('-')[0]
+
+            # build target manifest entry
+            target_manifest.append({})
+            target_manifest[target_i]['audio_filepath'] = file['audio_filepath']
+            target_manifest[target_i]['duration'] = file['duration']
+            target_manifest[target_i]['text'] = file['text']
+            target_manifest[target_i]['words'] = words
+            target_manifest[target_i]['alignments'] = end_times
+            target_manifest[target_i]['speaker_id'] = speaker_id
+
+            src_i += 1
+            target_i += 1
+
+        alignment_file.close()
+    write_manifest(output_path, target_manifest)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="LibriSpeech Alignment Manifest Creator")
+    parser.add_argument("--input_manifest_filepath", help="path to input manifest file", type=str, required=True)
```

Comment:
is this manifest file of librispeech audio samples? what are the keys script expecting?